source/common/stream_info/stream_info_impl.h

```diff
@@ -218,6 +219,25 @@ struct StreamInfoImpl : public StreamInfo {
     return upstream_transport_failure_reason_;
   }
 
+  void setRequestHeaders(const Http::HeaderMap& headers) override {
+    headers.iterate(
+        [](const Http::HeaderEntry& entry, void* self) {
+          static_cast<StreamInfoImpl*>(self)->request_headers_.emplace_back(
+              std::string(entry.key().getStringView()), entry.value().getStringView());
+          return Http::HeaderMap::Iterate::Continue;
+        },
+        this);
+  }
+
+  std::string getRequestHeader(const Http::LowerCaseString& name) const override {
```

Comment:
As a side-effect of removing the header copy, I think we can probably remove this method and just provide a method to return a `const Http::HeaderMap&` from the pointer. The header formatter can just use the HeaderMap interface to lookup the header.

Reply:
We should still consider this. There's probably some extra calls to std::string's copy-constructor happening.

Reply:
yep I've changed the return type to `const Http::HeaderMap*`